modules/templates/template_proxy_request.rb

```diff
@@ -19,8 +19,16 @@ def get_template(kind, env, params)
       # in hostgroup provisioning there are spaces
       kind = kind.split('/').map{|x| CGI.escape(x)}.join('/')
       logger.debug "Template: request for #{kind} using #{opts.inspect} at #{uri.host}"
-      proxy_headers = extract_request_headers(env).merge("X-Forwarded-For" => "#{env['REMOTE_ADDR']}, #{proxy_ip}")
-      proxy_req = request_factory.create_get("/unattended/#{kind}", opts, proxy_headers)
+      proxy_headers = extract_request_headers(env)
+      proxy_headers.merge!("X-Forwarded-For" => "#{env['REMOTE_ADDR']}, #{proxy_ip}")
```

Comment:
You can avoid merge here,

```suggestion
      proxy_headers['X-Forwarded-For'] = "#{env['REMOTE_ADDR']}, #{proxy_ip}"
```

Reply:
Copy pasted it, yeah :-)